File: alljoyn_objc/samples/iOS/BasicService/BasicService/ViewController.m
////////////////////////////////////////////////////////////////////////////////
//    Copyright (c) Open Connectivity Foundation (OCF), AllJoyn Open Source
//    Project (AJOSP) Contributors and others.
//
//    SPDX-License-Identifier: Apache-2.0
//
//    All rights reserved. This program and the accompanying materials are
//    made available under the terms of the Apache License, Version 2.0
//    which accompanies this distribution, and is available at
//    http://www.apache.org/licenses/LICENSE-2.0
//
//    Copyright (c) Open Connectivity Foundation and Contributors to AllSeen
//    Alliance. All rights reserved.
//
//    Permission to use, copy, modify, and/or distribute this software for
//    any purpose with or without fee is hereby granted, provided that the
//    above copyright notice and this permission notice appear in all
//    copies.
//
//    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
//    WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
//    WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
//    AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
//    DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR
//    PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER
//    TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
//    PERFORMANCE OF THIS SOFTWARE.
////////////////////////////////////////////////////////////////////////////////

#import "ViewController.h"
#import "BasicService.h"

@interface ViewController () <BasicServiceDelegate>

@property (nonatomic, strong) BasicService *basicService;
@property (nonatomic) BOOL activateService;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    _basicService = [[BasicService alloc] init];
    _basicService.delegate = self;
    [_basicService printVersionInformation];

    _activateService = YES;
    [_basicService startServiceAsync];
    [_serviceButton setTitle:@"Stop Service" forState:UIControlStateNormal];
}
- (IBAction)didTouchServiceButton:(id)sender {
    _activateService = !_activateService;
    if (_activateService) {
        [_basicService startServiceAsync];
        [_serviceButton setTitle:@"Stop Service" forState:UIControlStateNormal];
    } else {
        [_basicService stopServiceAsync];
        [_serviceButton setTitle:@"Start Service" forState:UIControlStateNormal];
    }
}

- (void)didReceiveStatusUpdateMessage:(NSString *)message
{
    dispatch_async(dispatch_get_main_queue(), ^{
        NSMutableString *string = self.eventsTextView.text.length ? [self.eventsTextView.text mutableCopy] : [[NSMutableString alloc] init];
        NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
        [formatter setTimeStyle:NSDateFormatterMediumStyle];
        [formatter setDateStyle:NSDateFormatterShortStyle];
        [string appendFormat:@"[%@] ",[formatter stringFromDate:[NSDate date]]];
        [string appendString:message];
        [self.eventsTextView setText:string];
        NSLog(@"%@", string);
    });
}

@end

File: alljoyn_objc/samples/iOS/Swift/BasicService/BasicService/ViewController.swift
//
//  ViewController.swift
//  BasicService
//
//  Created by Olga Konoreva on 30/11/16.
//  Copyright © 2016 AllJoyn. All rights reserved.
//

import UIKit

class ViewController: UIViewController, BasicServiceDelegate {

    @IBOutlet weak var eventsTextView: UITextView!

    let basicService = BasicService()

    override func viewDidLoad() {
        super.viewDidLoad()
        basicService.delegate = self
        basicService.startService()
    }

    func didReceiveStatusUpdateMessage(message: String) {

        print(message)

        DispatchQueue.main.async {
            var currentText = self.eventsTextView.text

            let date = NSDate()
            let calendar = NSCalendar.current
            let hour = calendar.component(.hour, from: date as Date)
            let minutes = calendar.component(.minute, from: date as Date)
            let seconds = calendar.component(.second, from: date as Date)

            currentText!.append("\(hour):\(minutes):\(seconds)\n")
            currentText!.append(message)

            self.eventsTextView.text = currentText
            print("\(currentText)")
        }
    }
}
